packages/roosterjs-content-model-plugins/lib/autoFormat/AutoFormatPlugin.ts

```diff
@@ -220,9 +220,14 @@ export class AutoFormatPlugin implements EditorPlugin {
                                 shouldFraction ||
                                 shouldOrdinals
                             );
-                        },
-                        formatOptions
+                        }
                     );
+                    if (formatOptions.changeSource) {
```

Comment:
What if instead of triggering another plugin event here we change the formatWithContentModel implementation by moving [line 27](https://github.com/microsoft/roosterjs/blob/f8a23bb492e1d08165f182dd27d6f9a3327d9cd4/packages/roosterjs-content-model-core/lib/coreApi/formatContentModel/formatContentModel.ts#L27) tbefore [line 81](https://github.com/microsoft/roosterjs/blob/f8a23bb492e1d08165f182dd27d6f9a3327d9cd4/packages/roosterjs-content-model-core/lib/coreApi/formatContentModel/formatContentModel.ts#L81C1-L89C15)

Since the option object is destructed to get the values and we modify the options object in the callback the AutoFormat Change source is not correctly set. 

@JiuqingSong to check if this is okay.

Reply:
Thanks, @BryanValverdeU!! That worked out better that my solution.